cloud/blockstore/libs/rdma/impl/client.cpp

```diff
@@ -1761,15 +1789,33 @@ void TClient::HandleConnectionEvent(NVerbs::TConnectionEventPtr event) noexcept
     }
 }
 
+void TClient::StopEndpoint(TClientEndpoint* endpoint) noexcept
```

Comment:
When you stop endpoint you call SetError which will invoke FlushQueues() and schedule reconnect.
Then reconnect is invoked and will call StopEndpoint destroying the QP and all the Send/Recv Buffers. But in theory there could be a race and the WR were not flushed yet. You need to first wait for all WR to be flushed and then invoke the destruction of qp and bufffers. You already have `TClientEndpoint::Flushed()` for this

Reply:
Good catch, I thought I did this, but forgot. Even mentioned it in https://github.com/ydb-platform/nbs/pull/1575#issuecomment-2228430254